weblate/api/serializers.py

```diff
@@ -507,6 +505,86 @@ class Meta:
         )
 
 
+class TermSerializer(serializers.ModelSerializer):
+    language = LanguageSerializer(required=False)
+
+    class Meta:
+        model = Term
+        fields = (
+            "language",
+            "id",
+            "source",
+            "target",
+        )
+
+    def create(self, validated_data):
+        language_validated = validated_data.get("language")
+        if language_validated:
+            validated_data["language"] = Language.objects.get(
+                code=language_validated.get("code")
+            )
+        term = Term.objects.create(**validated_data)
+        return term
```

Comment:
Handling of the source language attribute really should go to the  `to_internal_value` method. Handling it there makes all the validation have correct object instead of some intermediate values which fail the validation. This is not that big issue here, but I've hit the issue while creating https://github.com/WeblateOrg/weblate/pull/4458

Related code there is: https://github.com/WeblateOrg/weblate/pull/4458/files#diff-b6e5dde92daa37bd15834c3b38ed90b0R476